server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -731,9 +732,13 @@ public XContentBuilder snapshotsToXContent(final XContentBuilder builder, final
                 }
                 builder.endObject();
             }
-            final Version version = snapshotDetails.getVersion();
+            final IndexVersion version = snapshotDetails.getVersion();
             if (version != null) {
-                builder.field(VERSION, version.toString());
+                if (version.onOrBefore(IndexVersion.V_8_9_0)) {
```

Comment:
Backwards compatibility - Version string used for <8.9.0, id int used for >=8.9.0